components/brave_search/browser/backup_results_service.h

```diff
@@ -0,0 +1,66 @@
+// Copyright (c) 2025 The Brave Authors. All rights reserved.
+// This Source Code Form is subject to the terms of the Mozilla Public
+// License, v. 2.0. If a copy of the MPL was not distributed with this file,
+// You can obtain one at https://mozilla.org/MPL/2.0/.
+
+#ifndef BRAVE_COMPONENTS_BRAVE_SEARCH_BROWSER_BACKUP_RESULTS_SERVICE_H_
+#define BRAVE_COMPONENTS_BRAVE_SEARCH_BROWSER_BACKUP_RESULTS_SERVICE_H_
+
+#include <optional>
+#include <string>
+
+#include "base/functional/callback.h"
+#include "components/keyed_service/core/keyed_service.h"
+#include "net/http/http_request_headers.h"
+#include "url/gurl.h"
+
+namespace content {
+class WebContents;
+}
+
+namespace brave_search {
+
+// Fetches search results from a backup search provider,
+// for use in Brave Search fallback mixing (GMix) and Web Discovery Project.
+//
+// Each request will use an OTR profile for temporarily storing cookies, etc.
+//
+// There are three modes of operation for this service:
+// i)   If `features::kBackupResultsFullRender` is disabled, the initial search
+//      page will be rendered, and the actual search engine results page will be
+//      fetched.
+// ii)  If `features::kBackupResultsFullRender` is enabled, the initial search
+//      page and the actual search engine results page will be rendered.
```

Comment:
nit: it's not clear what we mean by "initial" vs "actual", can we just use the "fallback" search engine terminology?

Reply:
This whole class is in the context of a fallback engine, so i'm not sure if a "fallback page" would be accurate here. What do you think about "bootstrap page" and "real SERP page" or something similar?

Reply:
amended comments